gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/resource/integration/IntegrationResource.java

```diff
@@ -64,7 +74,46 @@ public Response getIntegrationById(@PathParam("integrationId") String integratio
             @Permission(value = RolePermission.ENVIRONMENT_API, acls = { RolePermissionAction.CREATE }),
         }
     )
-    public Response ingestApis(@PathParam("integrationId") String integrationId) {
-        return Response.ok(IntegrationIngestionResponse.builder().status(IngestionStatus.SUCCESS).build()).build();
+    public void ingestApis(@PathParam("integrationId") String integrationId, @Suspended final AsyncResponse response) {
+        var executionContext = GraviteeContext.getExecutionContext();
+        if (
+            !hasPermission(
+                executionContext,
+                RolePermission.ENVIRONMENT_INTEGRATION,
+                GraviteeContext.getCurrentEnvironment(),
+                RolePermissionAction.READ
+            ) ||
+            !hasPermission(
+                executionContext,
+                RolePermission.ENVIRONMENT_API,
+                GraviteeContext.getCurrentEnvironment(),
+                RolePermissionAction.CREATE
+            )
+        ) {
+            throw new ForbiddenAccessException();
+        }
```

Comment:
Do we need this check if we have @Permissions annotation on lines 71-76?

Reply:
When you have to check multiple permissions on the endpoint, the annotation is doing an OR. But in that case we need to do an AND that why I had to do that. We have the same kind of check in ApiResource for the duplicate operation